event/bench_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 The Go Authors. All rights reserved.
```

Comment:
can you provide some background on this event package?

Reply:
[`jsonrpc2_v2` is planned by the gopls team to go public](https://github.com/golang/go/issues/46187#issuecomment-842445007), but one of the dependencies it has is this event package. Both are internal as of now, so I had to duplicate them both for it to work. Is there a more elegant solution?